mzLib/MassSpectrometry/PeakIndexing/IndexedMassSpectralPeak.cs

```diff
@@ -6,16 +6,16 @@ namespace MassSpectrometry
     public class IndexedMassSpectralPeak : IIndexedPeak
     {
         public int ZeroBasedScanIndex { get; init; }
-        public double Mz { get; init; }
-        public double M => Mz;
-        public double RetentionTime { get; init; }
-        public double Intensity { get; init; }
+        public float Mz { get; init; }
+        public float M => Mz;
+        public float RetentionTime { get; init; }
+        public float Intensity { get; init; }
         public IndexedMassSpectralPeak(double mz, double intensity, int zeroBasedScanIndex, double retentionTime)
```

Comment:
You should change this constructor to take in floats instead of doubles. You'll also need to modify IndexingEngine to do the cast on the constructor call.

Alternately, add a new constructor that takes in floats